### PR TITLE
Library fixes

### PR DESCRIPTION
**Pull request by djcooke**

### CODE REVIEW

**Comment by djcooke** [comment]
I'll squash this into 2 commits after review
* library fixes
* naming scheme

File: miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultLibraryService.java [hunk 1]
```diff
@@ -537,11 +540,15 @@ private void validateAliasOrThrow(Library library) throws IOException {
...
-      } else {
-        // throw if duplicate aliases are not allowed (and in the case of a DetailedLibrary, if it has a standard alias
+    // duplicate aliases may be allowed via naming scheme, or with nonStandardAlias=true in the case of a DetailedLibrary
+    if (namingScheme.duplicateLibraryAliasAllowed()
+        || (LimsUtils.isDetailedLibrary(library) && ((DetailedLibrary) library).hasNonStandardAlias())) {
+      return;
+    }
+    List<Library> potentialDupes = listByAlias(library.getAlias());
+    for (Library potentialDupe : potentialDupes) {
+      if (library.getId() == AbstractLibrary.UNSAVED_ID || library.getId() != potentialDupe.getId()) {
```
Comment (callunity) [inline miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultLibraryService.java]: Should this change also be made in the `DefaultSampleService`?

File: miso-web/src/main/webapp/scripts/library_hot.js [hunk 2]
```diff
@@ -652,9 +654,16 @@ Library.hot = {
           lib.libraryDesignId = Hot.maybeGetProperty(Hot.findFirstOrNull(Hot.namePredicate(obj.libraryDesignAlias), Library.designs), 'id');
         }
         lib.libraryDesignCodeId = Hot.maybeGetProperty(Hot.findFirstOrNull(function (ldCode) { return ldCode.code == obj.libraryDesignCode; }, Hot.dropdownRef.libraryDesignCodes), 'id');
+      } else {
+        lib.type = 'Plain';
       }
 
-      lib.qcPassed = (obj.qcPassed && obj.qcPassed != 'unknown' ? obj.qcPassed : '') || '';
+      // lib.qcPassed = (obj.qcPassed && obj.qcPassed != 'unknown' ? obj.qcPassed : '') || '';
```
Comment (callunity) [inline miso-web/src/main/webapp/scripts/library_hot.js]: Please remove commented-out code.

Comment (apmasell) [inline miso-web/src/main/webapp/scripts/library_hot.js]: Please deleted commented code.

File: miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultLibraryService.java (hunk 1 above)
Comment (djcooke) [inline miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultLibraryService.java]: No. `DefaultSampleService#validateAliasUniqueness` is only called when a new sample is created or an existing sample's alias is changed

**Pull request merged**